File: 프로그래머스/lv2/42578. 위장/위장.kt
class Solution {
    fun solution(clothes: Array<Array<String>>): Int {
        var answer = 1
        val hash = HashMap<String, Int>()
        
        for(cloth in clothes){
            val key = cloth[1]
            
            if(hash.containsKey(key)){
                val v = hash[key]
                v?.let {
                    hash.put(key, v + 1)
                }
            }
            else {
                hash.put(key, 1)
            }
        }
        
        for(keyValue in hash){
            answer *= (keyValue.value + 1)
        }
        
        return answer - 1
    }
}

File: 프로그래머스/lv2/42578. 위장/위장.py
def solution(clothes):
    dic = {}
    for i, j in clothes:
        dic[j] = dic.get(j, 0) + 1 #옷 종류의 개수 넣어주기
        
    
    answer = 1
    for key in dic:
        answer *= (dic[key]+1)
        
    return answer - 1